compiler/rustc_const_eval/src/util/check_validity_requirement.rs

```diff
@@ -168,3 +168,29 @@ fn check_validity_requirement_lax<'tcx>(
 
     Ok(true)
 }
+
+pub(crate) fn validate_scalar_in_layout<'tcx>(
+    tcx: TyCtxtAt<'tcx>,
+    scalar: ScalarInt,
+    ty: Ty<'tcx>,
+) -> bool {
+    let machine = CompileTimeMachine::new(CanAccessMutGlobal::No, CheckAlignment::Error);
+
+    let typing_env = ty::TypingEnv::fully_monomorphized();
+    let mut cx = InterpCx::new(tcx.tcx, tcx.span, typing_env, machine);
+
+    let Ok(layout) = cx.layout_of(ty) else { return false };
```

Comment:
Reporting `false` in case of a layout computation error seems odd? This is almost certainly a bug, right?

Reply:
I could just unwrap it and we'll do it properly when we can write a test for it

Reply:
That works for me.
@matthiaskrgr heads-up if you see this ICE, please ping us and thanks in advance for generating a test case. :)